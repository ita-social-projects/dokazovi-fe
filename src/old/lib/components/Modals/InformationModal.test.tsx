import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {
  InformationModal,
  IConfirmationModalWithButtonProps,
} from './InformationModal';

const PROPS_MOCK: IConfirmationModalWithButtonProps = {
  message: 'Test Message',
  loading: true,
  disabled: false,
  onClose: jest.fn(),
  isOpen: true,
};

const renderModal = () => {
  render(
    <InformationModal
      message={PROPS_MOCK.message}
      onClose={PROPS_MOCK.onClose}
      isOpen={PROPS_MOCK.isOpen}
    />,
  );
};

describe('InformationModal', () => {
  beforeEach(() => renderModal());
  it('render component and match snapshot', () => {
    const { asFragment } = render(
      <InformationModal
        message={PROPS_MOCK.message}
        onClose={PROPS_MOCK.onClose}
        isOpen={PROPS_MOCK.isOpen}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('InformationModal events', () => {
  it('should closes when close button is clicked', () => {
    const { asFragment } = render(
      <InformationModal
        onClose={PROPS_MOCK.onClose}
        isOpen={PROPS_MOCK.isOpen}
        message={PROPS_MOCK.message}
      />,
    );
    const closeBtn = screen.getByText('Закрити');
    userEvent.click(closeBtn);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('InformationModal loading status', () => {
  it('render with loading icon', () => {
    const { asFragment } = render(
      <InformationModal
        message={PROPS_MOCK.message}
        onClose={PROPS_MOCK.onClose}
        isOpen={PROPS_MOCK.isOpen}
        loading
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
