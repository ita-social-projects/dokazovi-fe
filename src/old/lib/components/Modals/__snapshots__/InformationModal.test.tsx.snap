// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationModal events should closes when close button is clicked 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`InformationModal loading status render with loading icon 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`InformationModal render component and match snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
